src/main/java/bio/terra/service/snapshotbuilder/CriteriaQueryBuilder.java

```diff
@@ -0,0 +1,197 @@
+package bio.terra.service.snapshotbuilder;
+
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.model.SnapshotBuilderCriteria;
+import bio.terra.model.SnapshotBuilderCriteriaGroup;
+import bio.terra.model.SnapshotBuilderDomainCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataListCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataRangeCriteria;
+import bio.terra.service.snapshotbuilder.query.FieldPointer;
+import bio.terra.service.snapshotbuilder.query.FieldVariable;
+import bio.terra.service.snapshotbuilder.query.FilterVariable;
+import bio.terra.service.snapshotbuilder.query.Literal;
+import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.TablePointer;
+import bio.terra.service.snapshotbuilder.query.TableVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.BinaryFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.BooleanAndOrFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.FunctionFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.NotFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.SubQueryFilterVariable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.arrow.util.VisibleForTesting;
+
+public class CriteriaQueryBuilder {
+  public static final String ROOT_TABLE_NAME = "person";
```

Comment:
if we think the root table name will always be person, okay. If not, we could have a constructor that takes in this value.

Reply:
I think it will be configurable, but a constructor seems good.

Reply:
I think what we'd want to do is to put this in `SnapshotBuilderSettings` and pass that to the builder constructor